packages/hoppscotch-sh-admin/src/components/settings/AuthProvider.vue

```diff
@@ -35,25 +35,29 @@
               :key="field.key"
               class="mt-5"
             >
-              <label>{{ field.name }}</label>
-              <span class="flex">
-                <HoppSmartInput
-                  v-model="provider.fields[field.key]"
-                  :type="
-                    isMasked(provider.name, field.key) ? 'password' : 'text'
-                  "
-                  :disabled="isMasked(provider.name, field.key)"
-                  :autofocus="false"
-                  class="!my-2 !bg-primaryLight"
-                />
-                <HoppButtonSecondary
-                  :icon="
-                    isMasked(provider.name, field.key) ? IconEye : IconEyeOff
-                  "
-                  class="bg-primaryLight h-9 mt-2"
-                  @click="toggleMask(provider.name, field.key)"
-                />
-              </span>
+              <template
+                v-if="field.applicableProviders.includes(provider.name)"
+              >
+                <label>{{ field.name }}</label>
+                <span class="flex max-w-lg">
+                  <HoppSmartInput
+                    v-model="provider.fields[field.key as keyof typeof provider['fields']]"
+                    :type="
+                      isMasked(provider.name, field.key) ? 'password' : 'text'
+                    "
+                    :disabled="isMasked(provider.name, field.key)"
+                    :autofocus="false"
+                    class="!my-2 !bg-primaryLight flex-1"
+                  />
+                  <HoppButtonSecondary
+                    :icon="
+                      isMasked(provider.name, field.key) ? IconEye : IconEyeOff
+                    "
+                    class="bg-primaryLight h-9 mt-2"
+                    @click="toggleMask(provider.name, field.key)"
+                  />
+                </span>
+              </template>
```

Comment:
The fields are conditionally rendered after checking the current provider in iteration against the `applicableProviders` field.